### DISCUSSION TITLE
[solved] magick: no decode delegate for this image format `PANGO' @ error/constitute.c/ReadImage/741.

### DISCUSSION THREAD

**Discussion by htohneb** [ImageMagick/ImageMagick#6131] [Help] [answered]
### ImageMagick version

7.1.0-62 Q16-HDRI aarch64 20885

### Operating system, version and so on

mac os x 13.2.1 (compiler gcc 4.2)

### Description

Hello everyone,

maybe, pango is't correct installt?

What I have done in one terminal:

- I Install Imagemagick (7.1.0-62) with brew
- I Install Pango (1.50.14.arm64_ventura.bottle.tar.gz) with brew

> job@Touch Desktop % brew install imagemagick
> ==> Fetching dependencies for imagemagick: freetype
> ==> Fetching freetype
> ==> Downloading https://ghcr.io/v2/homebrew/core/freetype/manifests/2.13.0_1
> ######################################################################## 100.0%
> ==> Downloading https://ghcr.io/v2/homebrew/core/freetype/blobs/sha256:9dec5b349d68fd2925b84cda39af7842c29cb709bce44fb4b3e3cc0ed425eea5
> ==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:9dec5b349d68fd2925b84cda39af7842c29cb709bce44fb4b3e3cc0ed425eea5?se=2023-0
> ######################################################################## 100.0%
> ==> Fetching imagemagick
> ==> Downloading https://ghcr.io/v2/homebrew/core/imagemagick/manifests/7.1.0-62
> Already downloaded: /Users/job/Library/Caches/Homebrew/downloads/0c8e50cda2fbbf21324afa758b413925db3870ce7b42fc55144834fffd7e1301--imagemagick-7.1.0-62.bottle_manifest.json
> ==> Downloading https://ghcr.io/v2/homebrew/core/imagemagick/blobs/sha256:9077aa03701635fdd66026855db9c87e1c50c52c39c1b1fa831a08aad95b9d5d
> Already downloaded: /Users/job/Library/Caches/Homebrew/downloads/a1a6548da5aa9b0e16f217cb41e13732fb3ed8102d3699bb45e7f606de9b9395--imagemagick--7.1.0-62.arm64_ventura.bottle.tar.gz
> ==> Installing dependencies for imagemagick: freetype
> ==> Installing imagemagick dependency: freetype
> ==> Pouring freetype--2.13.0_1.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/freetype/2.13.0_1: 67 files, 2.4MB
> ==> Installing imagemagick
> ==> Pouring imagemagick--7.1.0-62.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/imagemagick/7.1.0-62: 807 files, 31MB
> ==> Running `brew cleanup imagemagick`...
> Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
> Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
> ==> Upgrading 1 dependent of upgraded formulae:
> Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
> Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
> harfbuzz 7.0.1 -> 7.1.0
> ==> Fetching harfbuzz
> ==> Downloading https://ghcr.io/v2/homebrew/core/harfbuzz/manifests/7.1.0
> ######################################################################## 100.0%
> ==> Downloading https://ghcr.io/v2/homebrew/core/harfbuzz/blobs/sha256:49464bacf4e6af3dfcbb6c7ad6b3837bff639ad7938909fd5ca412593a9203ff
> ==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:49464bacf4e6af3dfcbb6c7ad6b3837bff639ad7938909fd5ca412593a9203ff?se=2023-0
> ######################################################################## 100.0%
> ==> Upgrading harfbuzz
>   7.0.1 -> 7.1.0 
> 
> ==> Pouring harfbuzz--7.1.0.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/harfbuzz/7.1.0: 76 files, 8.8MB
> ==> Running `brew cleanup harfbuzz`...
> Removing: /opt/homebrew/Cellar/harfbuzz/7.0.1... (76 files, 8.8MB)
> Removing: /Users/job/Library/Caches/Homebrew/harfbuzz--7.0.1... (2.2MB)
> ==> Checking for dependents of upgraded formulae...
> ==> No broken dependents found!
> job@Touch Desktop % brew install pango      
> ==> Fetching pango
> ==> Downloading https://ghcr.io/v2/homebrew/core/pango/manifests/1.50.14
> ######################################################################## 100.0%
> ==> Downloading https://ghcr.io/v2/homebrew/core/pango/blobs/sha256:36b5b69c52886ea5c6599bc35bf22eb942cc44b2bcbe2ea0bd2340d72fe1d832
> ==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:36b5b69c52886ea5c6599bc35bf22eb942cc44b2bcbe2ea0bd2340d72fe1d832?se=2023-0
> ######################################################################## 100.0%
> ==> Pouring pango--1.50.14.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/pango/1.50.14: 68 files, 3.3MB
> ==> Running `brew cleanup pango`...
> Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
> Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
> Removing: /Users/job/Library/Caches/Homebrew/pango--1.50.13... (803.0KB)

Then I want to test it, with the command:

**_magick -background lightblue -gravity center -size 180x \
          pango:"The <b>bold</b> and <i>beautiful</i>" \
          pango_formatting.gif_**

From the side: [https://imagemagick.org/Usage/text/#pango](https://imagemagick.org/Usage/text/#pango)

What IM me say:

**magick: unable to open image 'The <b>bold</b> and <i>beautiful</i>': No such file or directory @ error/blob.c/OpenBlob/3569.
magick: no decode delegate for this image format `PANGO' @ error/constitute.c/ReadImage/741.**

magick --version:

> Version: ImageMagick 7.1.0-62 Q16-HDRI aarch64 20885 https://imagemagick.org
> Copyright: (C) 1999 ImageMagick Studio LLC
> License: https://imagemagick.org/script/license.php
> Features: Cipher DPC HDRI Modules OpenMP(5.0) 
> Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg jxl lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
> Compiler: gcc (4.2)

Have someone an idea?

Thank for reading, and your time

**Comment by urban-warrior** [comment]
Notice `pangocairo` is not listed as a delegate?  The Pango delegate library was not enabled at ImageMagick build time.  If you build ImageMagick yourself, you can check `config.log` to discover why the Pango delegate library failed to validate.

**Reply by htohneb** [reply]
Thank you [urban-warrior](https://github.com/urban-warrior) you'r fast!

The installation with brew, wasn't from the source, that was the package from brew.

ok,

now I deinstall with brew pango and IM.

The right installation from source, is this?

[http://www.gctv.ne.jp/~yokota/magick/](http://www.gctv.ne.jp/~yokota/magick/)

Thanks for your reply

**Reply by htohneb** [reply]
I Test now:

uninstall with brew from IM

Then install with **brew install imagemagick -S** (from source)

That change nothing.

and I don't found the **config.log** file from IM.

**Comment by fmw42** [comment]
You have to install Imagemagick and Pango at the same time and in the same brew command.  Search Google for how to install Imagemagick with delegates from Homebrew.

**Reply by htohneb** [reply]
Thanks fmw42, 

brew haven't an option, to install pango and Imagemagick, since 5 years. The option are deleted.

Now, I'll search another, engine for install, IM and Pango at the same command.

now I test macports with **sudo port install imagemagick +pango**

But Imagemagick installd, maybe, not reachable?

> job@Touch Desktop % magick -background lightblue pango:"Anthony Thyssen" pango.gif
> zsh: command not found: magick
> job@Touch Desktop % magick -version
> zsh: command not found: magick
> job@Touch Desktop % convert -version
> Version: ImageMagick 6.9.11-60 Q16 arm 2021-01-25 https://imagemagick.org
> Copyright: (C) 1999-2021 ImageMagick Studio LLC
> License: https://imagemagick.org/script/license.php
> Features: Cipher DPC Modules 
> Delegates (built-in): bzlib djvu fftw fontconfig freetype gslib heic jbig jng jp2 jpeg lcms ltdl lzma openexr pangocairo png ps raw tiff webp x xml zlib
> job@Touch Desktop % 
> 
Have someone a idea?

Thank you for reading and your time

**Comment by snibgo** [comment, marked as answer]
I know nothing about Macs or macports.

> Version: ImageMagick 6.9.11-60 Q16 arm 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules
Delegates (built-in): bzlib djvu fftw fontconfig freetype gslib heic jbig jng jp2 jpeg lcms ltdl lzma openexr pangocairo png ps raw tiff webp x xml zlib

You have `pangocairo`, so pango should work with that IM. Try this:
```
convert pango:"Hello world" x.png
```

**Comment by htohneb** [comment]
Wowowowowowowo

**_that works_**

Thank you [snibgo](https://github.com/snibgo), very much!
